Firmware/Pluma-Adafruit-BNO055/Pluma-Adafruit-BNO055.ino

```diff
@@ -0,0 +1,195 @@
+#include <Wire.h>
+#include <Adafruit_Sensor.h>
+#include <Adafruit_BNO055.h>
+#include <utility/imumaths.h>
+
+/* This driver uses the Adafruit unified sensor library (Adafruit_Sensor),
+   which provides a common 'type' for sensor data and some helper functions.
+
+   To use this driver you will also need to download the Adafruit_Sensor
+   library and include it in your libraries folder.
+
+   You should also assign a unique ID to this sensor for use with
+   the Adafruit Sensor API so that you can identify this particular
+   sensor in any data logs, etc.  To assign a unique ID, simply
+   provide an appropriate value in the constructor below (12345
+   is used by default in this example).
+
+   Connections
+   ===========
+   Connect SCL to analog 5
+   Connect SDA to analog 4
+   Connect VDD to 3.3-5V DC
+   Connect GROUND to common ground
+
+   History
+   =======
+   2015/MAR/03  - First release (KTOWN)
+*/
+
+/* Set the delay between fresh samples */
+//uint16_t BNO055_SAMPLERATE_DELAY_MS = 100;
+uint16_t BNO055_SAMPLERATE_DELAY_MS = 1;
+
+
+// Check I2C device address and correct line below (by default address is 0x29 or 0x28)
+//                                   id, address
+Adafruit_BNO055 bno = Adafruit_BNO055(55, 0x28, &Wire);
+
+//sync_sig_pin
+int sync_sig_pin=23;
+
+//LED pin
+int LED_pin=8;
+
+int my_val=0;
+
+// ----------------------------------------------------------------------------------------
+void setup(void)
+{
+  Serial.begin(115200);
+  pinMode(sync_sig_pin, INPUT);  
+  pinMode(LED_pin, OUTPUT);  
+  //digitalWrite(LED_pin,HIGH);
+
+  while (!Serial) delay(10);  // wait for serial port to open!
+
+  Serial.println("Orientation Sensor Test"); Serial.println("");
+
+  /* Initialise the sensor */
+  if (!bno.begin())
+  {
+    /* There was a problem detecting the BNO055 ... check your connections */
+    Serial.print("Ooops, no BNO055 detected ... Check your wiring or I2C ADDR!");
+    while (1);
+  }
```

Comment:
Should we keep these log and warning messages? The high-level interface might have trouble parsing these messages.

Reply:
The interface will crash with these messages. 

- The first message is part of the correct initialization of the device, so yes is to be removed.
- If there is an initialization error and  the device is not initialized, it should crash. Keeping that message is good for debugging.

Reply:
Fix in 48146acbdc0e50de8dfd0589051b3a48a394bf80